tda/loop.sh

```diff
@@ -52,10 +52,44 @@ trap 'log_signal "SIGIO" 29' IO
 trap 'log_signal "SIGPWR" 30' PWR
 trap 'log_signal "SIGSYS" 31' SYS
 
+# Day tracking file to persist the last day number
+DAY_TRACKING_FILE="/tmp/tda_loop_prev_day"
+
+# Function to get current day number
+get_current_day() {
+    date +%j
+}
+
+# Function to check if day has changed and set ONCE_PER_DAY accordingly
+check_day_change() {
+    local current_day=$(get_current_day)
+    local prev_day=""
+    
+    # Read the previous day number from file if it exists
+    if [ -f "$DAY_TRACKING_FILE" ]; then
+        prev_day=$(cat "$DAY_TRACKING_FILE" 2>/dev/null || echo "")
+    fi
+    
+    # If this is the first run or day has changed, set ONCE_PER_DAY=1
+    if [ -z "$prev_day" ] || [ "$current_day" != "$prev_day" ]; then
+        export ONCE_PER_DAY=1
```

Comment:
ONCE_PER_DAY took a little while to grammatically wrap my head around

If you want to make it more instantly understandable, consider `IS_FIRST_AUTOMATED_RUN_OF_DAY` or something, so it's more readable when used in conditionals

Reply:
thanks, good call